backend/core/src/core/history/application/history-service.ts

```diff
@@ -11,11 +11,22 @@ export class HistoryService {
     this.historyRepository = historyRepository;
   }
 
-  getAllUserHistory(userId: string): Promise<History[]> {
-    return this.historyRepository.getAllUserHistory(userId);
+  async getUserHistory(userId: string, page: string): Promise<HistoryResponseDto> {
+    const pageNumber = Number(page);
+    const lastPage = Math.ceil((await this.historyRepository.getAllUserHistoryLength(userId)) / 10);
+    if (!pageNumber) {
+      return this.historyRepository.getUserHistory(userId, 10, 0, lastPage);
+    }
+    const skip = (pageNumber - 1) * 10;
+    return this.historyRepository.getUserHistory(userId, 10, skip, lastPage);
```

Comment:
separate 10 into a variable, so that if you decide to change page size, you don't have to change it in 4 places

Reply:
done